### PR TITLE
Add support for functions.

### PR DESCRIPTION
**Pull request by bradhe**
This pull request extends the query syntax to apply functions to the result set, allowing the user to query and perform multiple operations at once. This implementation is very naive but gets this concept up off the ground.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -75,6 +75,56 @@ var _applyToPath = function(obj, path, val) {
   }
 };
 
+var doConcat = function(obj, args) {
```
Comment (cp) [inline src/index.js]: maybe it's worth just pulling lodash in for all this stuff? we can just import what we need.

Comment (woodburndesigns) [reply src/index.js]: I agree. There is no need to recreate the wheel everywhere.

Comment (bradhe) [reply src/index.js]: Wait, how does lodash help here? What would the lodash implementation look like?

Comment (cp) [reply src/index.js]: This `doConcat` function is actually querying JSONPath, but `_contains` could just use `lodash/concat`, and `doUniq` could use `lodash/uniq` I believe. @woodburndesigns?

Comment (woodburndesigns) [reply src/index.js]: I think arr.concat is actually fine in this case. But helpers like contains and unique could start coming out of lodash. Definitely not a blocker just don't think we need to continue writing our own helpers in each repo that pretty much do the same thing

Comment (bradhe) [reply src/index.js]: Updated to include lodash--take another peek?

**Comment by cp** [comment]
Love it. 👍 🚢

**Review by cp — APPROVED** [review]

**Pull request merged**